include/zephyr/sw_isr_table.h

```diff
@@ -77,6 +77,16 @@ void shared_isr(const void *data);
 extern struct shared_irq_data _shared_sw_isr_table[];
 #endif /* CONFIG_SHARED_INTERRUPTS */
 
+/**
+ * @brief Helper function used to compute the index in _sw_isr_table
+ * based on passed IRQ.
+ *
+ * @param irq IRQ number in its zephyr format
+ *
+ * @return corresponding index in _sw_isr_table
+ */
+unsigned int get_sw_isr_table_idx(unsigned int irq);
```

Comment:
This is a public header file. This function should be prefixed by something like `z_isr_`.

Reply:
I'm assuming this is because this function should only be used internally? Also, does the same logic apply to `shared_isr()`, which should be prefixed by a `_`?

Reply:
For now we stand by the "old" rule that non-public functions in public headers should be prefixed by `z_`. This is gonna change but it hasn't yet.

Reply:
Thanks for the clarification ! Is there also a rule regarding the naming of structures (i.e: should `struct shared_irq_data` be turned into `struct _shared_irq_data` like in the case of `struct _isr_list`)?

Reply:
The `old` rule is to not prefix with underscore. So `z_isr_*` would be better.

Reply:
Fixed. `get_sw_isr_table_idx` is now prefixed by a `z_`, while the shared interrupt data structures and functions are now prefixed by a `z_shared_isr` (except for the table, which has been renamed to `z_shared_sw_isr_table`)